.github/workflows/githubactions.yml

```diff
@@ -1,87 +1,258 @@
+        with:
+          lfs: true
+          submodules: recursive
+      
+      - name: Install pipenv
+        run: pip install pipenv==2021.5.29
+
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+          cache: pipenv
+          cache-dependency-path: Pipfile.lock
+
+      - name: Install CMake
+        uses: lukka/get-cmake@latest
+
+      - name: Install dependencies
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt-get install ccache valgrind
+          pipenv install
+
+      - name: Setup CCache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.mode }}-test
+
+      - name: Build
+        run: |
+          sudo /usr/sbin/update-ccache-symlinks
+          export PATH="/usr/lib/ccache:$PATH"
+          cmake -S boards -B $BUILD_DIR -DPLATFORM=x86 -DCMAKE_BUILD_TYPE=Release
+          make --directory=$BUILD_DIR -j2
+
+      - name: Run tests
+        run: |
+          cd $BUILD_DIR
+          ctest -T memcheck --verbose
+          cd -
+
+      - name: Check CCache statistics
+        run: ccache -s
+
+  build:
+    name: Build ARM Executables
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        board:
+          - FSM
+          - PDM
+          - BMS
+          - DCM
+          - DIM
+    env:
+      CCACHE_DIR: ${{ github.workspace }}/.ccache
+      BUILD_DIR: boards/arm_build
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: true
+          submodules: recursive
+      
+      - name: Install pipenv
+        run: pip install pipenv==2021.5.29
+
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+          cache: pipenv
+          cache-dependency-path: Pipfile.lock
+
+      - name: Install CMake
+        uses: lukka/get-cmake@latest
+
+      - name: Install dependencies
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt-get install ccache valgrind
+          pipenv install
+
+      - name: Setup XVFB
+        run: |
+          sudo cp ./scripts/environment_setup/xvfb.service /etc/systemd/system/xvfb.service
+          sudo systemctl start xvfb
+
+      # This should eventually be cached, but cannot be due to file permission issues
+      # https://github.com/actions/toolkit/issues/946
+      - name: Install STM32CubeMX
+        run: pipenv run sudo python scripts/environment_setup/install_cube.py /usr/local/STM32CubeMX ./tools/en.STM32CubeMX_v5-3-0.zip
+
+      - name: Setup CCache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.board }}-build
+
+      - name: Install ARM GCC
+        env:
+          # Unfortunately, the download link is a literal URL which means it does not
+          # pull the latest version automatically.
```

Comment:
probably best that the version (arm-none-eabi-9-2019-q4-major) of this is the same as our embedded builds? doesnt need to be the latest version

Reply:
It's using the same version that was in the TravisCI scripts. I just copied the surrounding comments as well.